src/client/auth/test.rs

```diff
@@ -70,3 +72,37 @@ async fn negotiate_mangled() {
         AuthMechanism::ScramSha1
     );
 }
+
+fn scram_sasl_first_options(mechanism: AuthMechanism) {
+    let sasl_first = SaslStart::new(String::new(), mechanism, Vec::new());
+    let command = sasl_first.into_command();
+    let options = match command.body.get_document("options") {
+        Ok(options) => options,
+        Err(_) => panic!("SaslStart should contain options document"),
+    };
+    match options.get_bool("skipEmptyExchange") {
+        Ok(skip_empty_exchange) => assert!(
+            skip_empty_exchange,
+            "skipEmptyExchange should be true for SCRAM authentication"
+        ),
+        Err(_) => panic!("SaslStart options should contain skipEmptyExchange"),
+    }
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn sasl_first_options_specified() {
+    scram_sasl_first_options(AuthMechanism::ScramSha1);
+    scram_sasl_first_options(AuthMechanism::ScramSha256);
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg(feature = "tokio-runtime")]
+async fn sasl_first_options_not_specified() {
```

Comment:
Now that X.509 is merged, could you update this to use that instead of AWS so we don't have to make this test tokio-specific?